### PR TITLE
feat(rna): add Authenticator component

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add base `Authenticator` component and unit tests

#### Issue #, if available
NA

#### Description of how you validated changes
Manually testing, unit testing
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-native/src/Authenticator/__tests__/Authenticator.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        () => ({ route } as unknown as UseAuthenticator)
+      );
+
+      const { toJSON } = render(
+        <Authenticator>
+          <TestChildren />
+        </Authenticator>
+      );
+
+      expect(toJSON()).toMatchSnapshot();
```
Comment (ioanabrooks) [inline packages/react-native/src/Authenticator/__tests__/Authenticator.spec.tsx:77]: Any other assertions we want to add, besides snapshot testing? Snapshots offer no guarantees that appearance or behavior is correct/as expected.

Can we also add some tests for unhappy path or any edge cases?

File: packages/react-native/src/Authenticator/Authenticator.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  UseAuthenticator,
+} from '@aws-amplify/ui-react-core';
+
+import { AuthenticatorProps, FormHandlers } from './types';
+
+const DEFAULTS = {
+  // TODO: remove cast after adding missing components
+} as unknown as AuthenticatorComponentDefaults<FormHandlers>;
+
+const isAuthenticatedRoute = (route: UseAuthenticator['route']) =>
```
Comment (ioanabrooks) [inline packages/react-native/src/Authenticator/Authenticator.tsx:20]: Would this be better placed in a separate utils file?

Comment (calebpollman) [reply packages/react-native/src/Authenticator/Authenticator.tsx:20]: If we need it in other places we can abstract it out but since it's such a minimal util think it's fine here for now

File: packages/react-native/src/Authenticator/__tests__/Authenticator.spec.tsx (hunk 1 above)
Comment (calebpollman) [reply packages/react-native/src/Authenticator/__tests__/Authenticator.spec.tsx:77]: Since we are just checking to make sure that the `children` argument is rendered for these `route` values, think we are okay with the snapshot but curious what other assertions/edge cases you are thinking?

Comment (ioanabrooks) [reply packages/react-native/src/Authenticator/__tests__/Authenticator.spec.tsx:77]: I was thinking to give the Child component a testID:
```
<Text testId="testChild">Test Children</Text>
```
and then actually checking that it exists in the tree. snapshot testing is only good for checking if UI changed unexpectadly, it doesn't tell you the child rendered.

Comment (ioanabrooks) [reply packages/react-native/src/Authenticator/__tests__/Authenticator.spec.tsx:77]: For the edge cases nothing particular off the top of my head, maybe like if you pass in an invalid component as child, does it blow up? Things liek that.

File: packages/react-native/src/Authenticator/Authenticator.tsx (hunk 2 above)
Comment (ioanabrooks) [reply packages/react-native/src/Authenticator/Authenticator.tsx:20]: How would one discover that this util exists here and not duplicate it in another component?

Comment (calebpollman) [reply packages/react-native/src/Authenticator/Authenticator.tsx:20]: How would it living in a _util.ts_ as a sibling to _Authenticator.tsx_ component be any different? 

If we really want to think about whether this will be reused, we are not exposing `route` anywhere else in the UI besides in `Authenticator`, so unless we need to suddenly change the architecture of the functional components, there is really no chance that this util will ever be reused

File: packages/react-native/src/Authenticator/__tests__/Authenticator.spec.tsx (hunk 1 above)
Comment (calebpollman) [reply packages/react-native/src/Authenticator/__tests__/Authenticator.spec.tsx:77]: > snapshot testing is only good for checking if UI changed unexpectedly, it doesn't tell you the child rendered.

But the child is what the snapshot contains right?

> maybe like if you pass in an invalid component as child, does it blow up?

Is that a test we should write? Like at that point we are just testing the React Native library itself right?

File: packages/react-native/src/Authenticator/Authenticator.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  route === 'authenticated' || route === 'signOut';
```
Comment (joebuono) [inline packages/react-native/src/Authenticator/Authenticator.tsx:21]: Are these the only two routes, or will we add more? If there's more, maybe we can put the routes in an array and use the `.includes` method

**Review by joebuono — APPROVED** [review]

File: packages/react-native/src/Authenticator/__tests__/Authenticator.spec.tsx (hunk 1 above)
Comment (ioanabrooks) [reply packages/react-native/src/Authenticator/__tests__/Authenticator.spec.tsx:85]: thanks for adding assertions!

**Review by ioanabrooks — APPROVED** [review]
LGTM

**Review by joebuono — APPROVED** [review]
